frontend/web/components/pages/GitHubSetupPage.tsx

```diff
@@ -0,0 +1,245 @@
+import React, { FC, useState, useEffect } from 'react'
+import OrganisationSelect from 'components/OrganisationSelect'
+import Input from 'components/base/forms/Input'
+import Icon from 'components/Icon'
+import InputGroup from 'components/base/forms/InputGroup'
+import ProjectFilter from 'components/ProjectFilter'
+import Utils from 'common/utils/utils'
+import Button from 'components/base/forms/Button'
+import { useCreateGithubIntegrationMutation } from 'common/services/useGithubIntegration'
+import { useCreateGithubRepositoryMutation } from 'common/services/useGithubRepository'
+import { useGetGithubReposQuery } from 'common/services/useGithub'
+import PanelSearch from 'components/PanelSearch'
+
+type Location = {
+  search: string
+}
+
+type GitHubSetupPageType = {
+  location: Location
+}
+
+type projectType = {
+  value: string
+  label: string
+}
+
+type repoType = {
+  value: string
+  label: string
+  name?: string
+}
+
+const GitHubSetupPage: FC<GitHubSetupPageType> = (props) => {
+  const installationId =
+    new URLSearchParams(props.location.search).get('installation_id') || ''
+  const [organisation, setOrganisation] = useState<string>('')
+  const [project, setProject] = useState<any>({})
+  const [projects, setProjects] = useState<projectType[]>([])
+  const [repositoryName, setRepositoryName] = useState<string>('')
+  const [repositoryOwner, setRepositoryOwner] = useState<string>('')
+  const [repositories, setRepositories] = useState<any>([])
+
+  const { data: repos, isSuccess: reposLoaded } = useGetGithubReposQuery(
+    {
+      installation_id: installationId,
+    },
+    { skip: !installationId },
+  )
+
+  const [
+    createGithubIntegration,
+    { data, isSuccess: isSuccessCreateGithubIntegration },
+  ] = useCreateGithubIntegrationMutation()
+  const baseUrl = window.location.origin
+
+  const [
+    createGithubRepository,
+    { isSuccess: isSuccessCreatedGithubRepository },
+  ] = useCreateGithubRepositoryMutation()
+
+  useEffect(() => {
+    if (reposLoaded && repos.repositories) {
+      setRepositoryOwner(repos?.repositories[0].owner.login)
+      setRepositories(repos)
+    }
+  }, [repos, reposLoaded])
+
+  useEffect(() => {
+    if (isSuccessCreatedGithubRepository) {
+      window.location.href = `${baseUrl}/`
+    }
+  }, [isSuccessCreatedGithubRepository])
+
+  useEffect(() => {
+    const createRepositories = async () => {
+      try {
+        if (data && isSuccessCreateGithubIntegration) {
+          const promises = []
+          for (const project of projects) {
+            const promise = createGithubRepository({
+              body: {
+                project: project.value,
+                repository_name: repositoryName,
+                repository_owner: repositoryOwner,
+              },
+              github_pk: data.id,
+              organisation_pk: organisation,
```

Comment:
organisation_pk is just the org id right? Is it possible to just replace these instances with organisation_id, I think requests.ts normally use this